docs/opentracing.rst

```diff
@@ -0,0 +1,179 @@
+===========
+Opentracing
+===========
+
+Background
+----------
+
+Opentracing is semi-standard being addopted by a number of distributed tracing
+platforms. It is a standardised api for facilitating vendor agnostic tracing
+instrumentation. That is, we can use the opentracing api and select one of a
+number of tracer implementations to do the heavy lifting in the background.
+Our current selected implementation is Jaeger.
+
+Opentracing concepts can be found at
```

Comment:
I'd find a very high-level overview of how it works useful here, to save me having to go and find the relevant bits in the upstream documentation. Just a sentence or two, like 

> Opentracing works by adding a header to HTTP requests, which identifies a 'span context', and whether it should be 'sampled'. This means that a single "span" can be traced between servers. It is up to the server what it does with the information it collects when it samples a span; in Synapse's case, it will report it to a Jaeger server.

(please edit for correctness!)

Reply:
I've changed this a lot.